public/app/features/explore/Explore.tsx

```diff
@@ -160,7 +161,7 @@ export class Explore extends React.PureComponent<ExploreProps, ExploreState> {
     }
   }
 
-  async setDatasource(datasource: DataSource) {
+  async setDatasource(datasource: DataSource & DataSourceApi, origin?: DataSource) {
```

Comment:
The typing here looks strange, DataSourceApi is not a subtype of DataSource.

Reply:
looking at the code, origin is also a DataSourceApi type

Reply:
separate note,  would you be in favor of rename DataSource to DataSourceDTO, and DataSourceApi to DataSource ?

Reply:
I'd rather remote typing here as long as types are not clearly defined.

Reply:
that's fine, or just add missing properties, they will evolve.